src/clincoded/static/components/dashboard.js

```diff
@@ -92,17 +92,40 @@ var Dashboard = createReactClass({
         }
         // Determine whether the classification had been previously approved
         if (snapshots && snapshots.length) {
-            // Only interested in knowing the presence of any "Approved" classification
-            filteredSnapshots = snapshots.filter(snapshot => snapshot.approvalStatus === 'Approved');
             // The "In progress" label shouldn't be shown after any given number of Provisional/Approval had been saved
             let sortedSnapshots = sortListByDate(snapshots, 'date_created');
+            // Only interested in knowing the presence of any "Approved" classification
+            filteredSnapshots = sortedSnapshots.filter(snapshot => snapshot.approvalStatus === 'Approved');
+
+            // Construct a label/tag to indicate the publish status of a GDM or Interpretation
+            let publishedStatusTag = null;
+            const publishedWarningMessage = (
+                <span className="publish-warning" data-toggle="tooltip" data-placement="top"
+                    data-tooltip="The current approved Classification is more recent than this published Classification.">
+                    <i className="icon icon-exclamation-triangle"></i>
+                </span>
+            );
+
+            for (let i = 0; i < filteredSnapshots.length; i++) {
```

Comment:
@bryanwulf, how about changing this line to `for (let snapshot of filteredSnapshots) {...}`?

Reply:
@jimmyzhen Since I'm checking the index within the loop (and it's not readily available when looping via for..of), I left this code as-is.